README
App-YTDL
========


INFO

Download single YouTube videos or choose videos from playlists/channels.
Before the download the script shows video infos and
lets you choose the video quality from the available qualities.
Instead of choosing the quality manually it is possible
to set and use preferred qualities.



INSTALLATION

    perl Makefile.pl
    make
    make test
    make install



HELP

yt-download -h




LICENSE AND COPYRIGHT

Copyright (C) 2013-2014 Kuerbis

This program is free software; you
can redistribute it and/or modify it under the same terms
as Perl 5.10.0. For details, see the full text of the
licenses in the file LICENSE.


Changes
Revision history for App-YTDL


0.029   2015-01-03
        - Removed option "auto_width" (now always on).
        - Added new option "Extractor directory".
        - Option "Logging" disabled by default.
        - Changed the default video directory.
        - Code refactoring.
        - Update documentation.

0.028   2014-12-30
        - Bugfix status message in "download_video".

0.027   2014-12-30
        - "Delete","Redo","Append": fixed video count.
        - Modified status messages.

0.026   2014-12-30
        - Bugfix: "sticky".
        - Bugfix: entries history file.
        - Code refactoring.

0.025   2014-12-29
        - Bugfix: ad channel to history only if youtube.
        - Code refactoring.

0.024   2014-12-27
        - New option "Channel directory".
        - Bug fixes.
        - Code refactoring.
        - Update documentation.

0.023   2014-12-25
        - Channels in the history file can be made sticky.
        - Update documentation.

0.022   2014-12-04
        - Bugfix: use "List::MoreUtils" for any and none.

0.021   2014-11-26
        - New option "Channel history".
        - New option "Sort order".
        - Code refactoring.
        - "Term::Choose" minimum version is "1.117".
        - Update documentation.

0.020_01   2014-11-01
        - Don't reset the previous selected videos if "FILTER" is called.
        - "Term::Choose" minimum version is "1.116_01".
        - Update documentation.

0.020   2014-09-27
        - New option "timeout".
        - Ask user what to do if error while fetching download info.

0.019   2014-09-14
        - Added "Term::Choose::Util" minimum version.

0.018   2014-09-10
        - Bugfixes.
        - Code refactoring.
        - Made "Term::Twiddle" a prerequisite if OS is not Win32.
        - Update documentation:

0.017   2014-08-23
        - Bugfix lists.
        - Perl minimum version from "5.10.1" to "5.10.0".

0.016   2014-08-07
        - Check if "youtube-dl" is available.
        - Added a filter for lists.
        - Output fixes and updates.

0.015   2014-07-21
        - Bugfixes and updates.

0.014   2014-07-21
        - "Set auto quality": added an "always default (best)" option.
        - Update documentation.

0.013   2014-07-21
        - "App::YTDL" uses now "youtube-dl" the get the data required for the video download.
        - Code refactoring.
        - Update Makefile.PL.
        - Update documentation.

0.012   2014-04-22
        - Code refactoring.
        - Out of order.

0.011   2014-03-10
        - "Term::Choose" version 1.100 or higher is required.
        - MSWin32 OS: it is required "Term::Choose" instead of "Term::Choose::Win32".
        - Experimental option "Video directory".
        - Update documentation.

0.010   2014-02-22
        - Bugfix: if OS is "MSWin32" -> term-width - 1 (windows console behaves differently).

0.009   2014-02-13
        - Update exit messages.
        - Update documentation.

0.008   2014-02-02
        - Fixed bug: moved "decode_argv()" to the beginning of the BEGIN block.
        - Code refactoring.

0.007   2014-02-01
        - Handle empty entrynodes.

0.006   2014-02-01
        - Added LICENSE file.
        - Detect encodings.
        - Handle not mappable characters.
        - Code refactoring.
        - Update documentation.

0.005   2014-01-26
        - Fixed precedence bug.

0.004   2014-01-24
        - Removed "channel" regexp.
        - Update "Auto width".
        - Update license.

0.003   2014-01-18
        - Added option "Auto width".
        - Update documentation.

0.002   2014-01-15
        - Fixed "channel" regexp.
        - Added option "UserAgent".
        - Updated documentation.

0.001   2014-01-12
        - First relase on CPAN.


Makefile.PL
use 5.010000;
use warnings;
use strict;
use ExtUtils::MakeMaker;


my %prereqs_os_specific;

if ( $^O eq 'MSWin32' ) {
    %prereqs_os_specific = (
        'Win32::Console::ANSI' => 0,
    );
}


WriteMakefile(
    PL_FILES         => {},
    MIN_PERL_VERSION => '5.010000',
    LICENSE          => 'perl',
    META_MERGE => {
        'meta-spec' => { version => 2 },
        resources   => {
            repository => {
              type => 'git',
              web  => 'https://github.com/kuerbis/App-YTDL',
              url  => 'https://github.com/kuerbis/App-YTDL.git',
            },
        },
    },
    AUTHOR           => 'Kuerbis <cuer2s@gmail.com>',
    NAME             => 'App::YTDL',
    VERSION_FROM     => 'lib/App/YTDL.pm',
    EXE_FILES        => [ 'bin/yt-download' ],
    BUILD_REQUIRES   => {
        'Test::More'             => 0,
    },
    PREREQ_PM => {
        'Encode'                 => 0,
        'Encode::Locale'         => 0,
        'Exporter'               => 0,
        'Fcntl'                  => 0,
        'File::Basename'         => 0,
        'File::HomeDir'          => 0,
        'File::Spec::Functions'  => 0,
        'File::Temp'             => 0,
        'File::Which'            => 0,
        'FindBin'                => 0,
        'Getopt::Long'           => 0,
        'IPC::System::Simple'    => 0,
        'JSON'                   => 0,
        'List::MoreUtils'        => 0,
        'List::Util'             => 0,
        'LWP::UserAgent'         => 0,
        'Pod::Usage'             => 0,
        'Term::ANSIScreen'       => 0,
        'Term::Choose'           => '1.117',
        'Term::Choose::Util'     => '0.016',
        'Term::ReadLine::Simple' => 0,
        'Text::LineFold'         => 0,
        'Time::HiRes'            => 0,
        'Try::Tiny'              => 0,
        'Unicode::GCString'      => 0,
        'Unicode::Normalize'     => 0,
        'URI'                    => 0,
        'URI::Escape'            => 0,
        'XML::LibXML'            => 0,
        %prereqs_os_specific,
    },
    dist  => {
        COMPRESS => 'gzip',
        SUFFIX   => '.gz',
        TAR      => 'tar',
        TARFLAGS => '--format=gnu -cvf',
    },
    clean => {
        FILES => 'App-YTDL-*',
    },
);
